manager/interfaces.go

```diff
@@ -89,3 +89,14 @@ type WriteKeypairFunc func(meta metadata.Metadata, key crypto.PrivateKey, chain
 // volume being published. Useful for modifying clients to make use of CSI
 // token requests.
 type ClientForMetadataFunc func(meta metadata.Metadata) (cmclient.Interface, error)
+
+// ReadyToRequestFunc can be optionally implemented by drivers to indicate whether
+// the driver is ready to request a certificate for the given volume/metadata.
+// This can be used to 'defer' fetching until later pod initialization events have
+// happened (e.g. CNI has allocated an IP if you want to embed a pod IP into the certificate
+// request resources).
+type ReadyToRequestFunc func(meta metadata.Metadata) bool
```

Comment:
If we don't do this now, then it would be a breaking change in future to add it. Although we don't have non-breaking guarantees, we should limit them where possible. I think we should add this :+1: 

For now we can just log the reason.